coins/monero/src/wallet/mod.rs

```diff
@@ -167,8 +176,13 @@ impl Scanner {
   /// burning_bug is a HashSet of used keys, intended to prevent key reuse which would burn funds.
   /// When an output is successfully scanned, the output key MUST be saved to disk.
   /// When a new scanner is created, ALL saved output keys must be passed in to be secure.
-  /// If None is passed, a modified shared key derivation is used which is immune to the burning
-  /// bug (specifically the Guaranteed feature from Featured Addresses).
+  /// If None is passed, a modified shared key derivation is used to generate new addresses
+  /// through this scanner which is immune to the burning bug (specifically
+  /// the Guaranteed feature from Featured Addresses).
```

Comment:
The original section had clearer English. If you want to highlight the address implications, how about adding
"This will effect generated/supported addresses."?